numpy-stubs/__init__.pyi

```diff
@@ -337,47 +345,51 @@ class ndarray(_ArrayOrScalarCommon, Iterable, Sized, Container):
     def dump(self, file: str) -> None: ...
     def dumps(self) -> bytes: ...
     def astype(
-        self,
+        self: _ArraySelf,
         dtype: _DtypeLike,
         order: str = ...,
         casting: str = ...,
         subok: bool = ...,
         copy: bool = ...,
-    ) -> ndarray: ...
-    def byteswap(self, inplace: bool = ...) -> ndarray: ...
-    def copy(self, order: str = ...) -> ndarray: ...
+    ) -> _ArraySelf: ...
+    def byteswap(self: _ArraySelf, inplace: bool = ...) -> _ArraySelf: ...
+    def copy(self: _ArraySelf, order: str = ...) -> _ArraySelf: ...
     @overload
-    def view(self, dtype: Type[_NdArraySubClass]) -> _NdArraySubClass: ...
+    def view(self, type: Type[_NdArraySubClass]) -> _NdArraySubClass: ...
     @overload
-    def view(self, dtype: _DtypeLike = ...) -> ndarray: ...
+    def view(self: _ArraySelf, dtype: _DtypeLike = ...) -> _ArraySelf: ...
     @overload
     def view(
         self, dtype: _DtypeLike, type: Type[_NdArraySubClass]
     ) -> _NdArraySubClass: ...
-    @overload
-    def view(self, *, type: Type[_NdArraySubClass]) -> _NdArraySubClass: ...
```

Comment:
This particular overload is redundant.
The signature described here expects ``type`` as a keyword-only argument, yet the very first overload already expects ``type`` as a keyword-or-positional argument.